src/App/App.tsx

```diff
@@ -16,22 +16,55 @@ const App = () => {
         }
         return initialTodoList
     })
+    useEffect(()=>localStorage.setItem('todos', JSON.stringify(todoList)), [todoList])
 
+    // AddItem
+    const addItem = useCallback((text:string) => {
+        const newArray = [
+            ...todoList,
+            {id: Date.now(), label: text, done: false},
+        ]
+        setTodoList(newArray)
+    }, [todoList])
+    
+    //Options
     const [filter, setFilter] = useState(FilterOptions.All)
-    const doneCount = todoList && todoList.filter((el: Element) => !el.done).length;
+    const doneCount = useMemo(()=>todoList && todoList.filter((el: Element) => !el.done).length,[todoList])
   
-    const setTodoListWithLocalStorage = useCallback((e: Array<Element>) => {
-        setTodoList(e)
-        localStorage.setItem('todos', JSON.stringify(e));
-    }, [])
+    // TodoList
+    const removeItem = useCallback((index:number) => {
+        const newArray = todoList.filter((el: Element) => el.id !== index);
+        setTodoList(newArray)
```

Comment:
it is better to use a function as an argument for `setTodoList`
> If the new state is computed using the previous state, you can pass a function to setState
You'll see the benefits of this ))

Reply:
```
    // TodoList
    const removeItem = useCallback((index:number) => {
        setTodoList((prevState)=>prevState.filter((el) => el.id !== index))
    }, [todoList])
```
I am not sure if that's what it's supposed to be.